crates/ruma-common/src/identifiers/room_id.rs

```diff
@@ -61,8 +61,12 @@ impl RoomId {
     }
 
     /// Create a `matrix.to` URI for an event scoped under this room ID.
-    pub fn matrix_to_event_uri(&self, ev_id: &EventId) -> MatrixToUri {
-        MatrixToUri::new((self, ev_id).into(), Vec::new())
+    pub fn matrix_to_event_uri<'a>(
+        &self,
+        ev_id: &EventId,
+        via: impl IntoIterator<Item = &'a ServerName>,
```

Comment:
You'd have to do:

```rust
let v = Vec<OwnedServerName>;
room_id.matrix_to_event_uri(v.iter().map(Deref::deref));
```

Reply:
We could also use `Item = T` where `T: AsRef<ServerName>`. Not sure..

Reply:
We could also `impl From<Owned*> for MatrixId` and `Item = T` where `T: Into<OwnedServerName>`, that would avoid unnecessary cloning.

Reply:
Sounds good, want to implement that? (then it would probably also make sense to replace all `&EventId` in these methods with `impl Into<OwnedEventId>`).

Reply:
> We could also `impl From<Owned*> for MatrixId`

Not sure why we havent done that before... oh well.

Reply:
Sure I'll do that then